weedcoco/importers/masks2.py

```diff
@@ -0,0 +1,125 @@
+from PIL import Image, ImageDraw
+import numpy as np
+from skimage import measure
+from shapely.geometry import Polygon, MultiPolygon
+
+
+def create_sub_masks(mask_image):
+    width, height = mask_image.size
+
+    # Initialize a dictionary of sub-masks indexed by RGB colors
+    sub_masks = {}
+    for x in range(width):
+        for y in range(height):
+            # Get the RGB values of the pixel
+            pixel = mask_image.getpixel((x, y))[:3]
+
+            # If the pixel is not black...
+            if pixel != (0, 0, 0):
+                # Check to see if we've created a sub-mask...
+                pixel_str = str(pixel)
+                sub_mask = sub_masks.get(pixel_str)
+                if sub_mask is None:
+                    # Create a sub-mask (one bit per pixel) and add to the dictionary
+                    # Note: we add 1 pixel of padding in each direction
+                    # because the contours module doesn't handle cases
+                    # where pixels bleed to the edge of the image
+                    sub_masks[pixel_str] = Image.new("1", (width + 2, height + 2))
+
+                # Set the pixel value to 1 (default is 0), accounting for padding
+                sub_masks[pixel_str].putpixel((x + 1, y + 1), 1)
+
+    return sub_masks
+
+
+def create_sub_mask_annotation(
+    sub_mask, image_id, category_id, annotation_id, is_crowd
+):
+    # Find contours (boundary lines) around each sub-mask
+    # Note: there could be multiple contours if the object
+    # is partially occluded. (E.g. an elephant behind a tree)
+    contours = measure.find_contours(sub_mask, 0.5, positive_orientation="low")
```

Comment:
Once you've got a boolean mask array, why not just use pycocotools.mask.encode to get a valid COCO segmentation out?
![image](https://user-images.githubusercontent.com/78827/95952604-28ccaf80-0e44-11eb-89c8-5cb5ab6d5f48.png)

Reply:
Ok you weren't kidding that there was a much simpler issue. I think my attempt to make it simpler ended up even further over engineering it...